integration-tests/build_run_cluster.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+echo $DRUID_INTEGRATION_TEST_OVERRIDE_CONFIG_PATH
+
+export DIR=$(cd $(dirname $0) && pwd)
+export HADOOP_DOCKER_DIR=$DIR/../examples/quickstart/tutorial/hadoop/docker
+export DOCKERDIR=$DIR/docker
+export SERVICE_SUPERVISORDS_DIR=$DOCKERDIR/service-supervisords
+export ENVIRONMENT_CONFIGS_DIR=$DOCKERDIR/environment-configs
+export SHARED_DIR=${HOME}/shared
+export SUPERVISORDIR=/usr/lib/druid/conf
+export RESOURCEDIR=$DIR/src/test/resources
+
+# so docker IP addr will be known during docker build
+echo ${DOCKER_IP:=127.0.0.1} > $DOCKERDIR/docker_ip
+
+if !($DRUID_INTEGRATION_TEST_SKIP_BUILD_DOCKER); then
+  sh ./script/copy_resources.sh
+  sh ./script/docker_build_containers.sh
+fi
+
+if !($DRUID_INTEGRATION_TEST_SKIP_RUN_DOCKER); then
+  sh ./stop_cluster.sh
+  sh ./script/docker_run_cluster.sh
+fi
+
+if ($DRUID_INTEGRATION_TEST_START_HADOOP_DOCKER); then
+  sh ./script/copy_hadoop_resources.sh
```

Comment:
copy_hadoop_resources.sh have to be run after hadoop cluster started BUT BEFORE the other druid containers start.

Reply:
I dont understand why we do this before running all containers.
We only add more delay to wait for containers.
Could you please explain why do we need to create all folders and other things before running all containers?

Reply:
Sure. Once you have started the Hadoop containers, hadoop will create a bunch of configuration files at /usr/local/hadoop/etc/hadoop/*.xml on the hadoop containers. These files will then have to be copy to the shared folder. Then our Druid java process will need to have these files in it's classpath when it startup. These cannot be added while Druid is already running, hence, that's why Druid processes are started after Hadoop

Reply:
If this files can't be added after Druid started then why I dont have any errors on running integration tests with hadoop?

Reply:
Can you verify if the hadoop tests still work if before you run the hadoop integration test your shared folder is empty. Since we never clean up the shared directory after the tests finish running, I think the reason you don't have any error is because your Druid cluster is picking up old files in .../shared/hadoop_xml from your previous runs.

Reply:
verified:
1. deleted ~/shared/hadoop_xml/*
2. s3 tests passed

Reply:
I think you want to run hdfs-deep-storage group since that test index_hadoop task.

Reply:
In case anyone is looking at this,
The hdfs-deep-storage test pass as is in this current patch.
I suspect that while we do start Druid before the hadoop_xml files get created, we do not run the ingestion task yet.
We do know where the files will be created so the classpath was set correctly when Druid starts. Hence, when we run the hadoop index ingestion task in the integration tests, those hadoop_xml files was created already.